### DISCUSSION TITLE
Consider using unoptimized IR locally

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#933] [Ideas]
Basically, enable via IR but disable the optimizer in the `lite` profile. As per @cameel's suggestion:

<img width="932" alt="SCR-20240523-sedj" src="https://github.com/sablier-labs/v2-core/assets/8782666/95318ffc-14cd-4bfc-b3d9-9b8e9ff2c129">

**Comment by smol-ninja** [comment]
I ran a few experiments and here are the results for `v2-core`.

### 1. Building only `src`

| Build time | $\text{ir = true}$ | $\text{ir = false}$ |
| - | - | - |
| $\text{optimizer = true}$ | 1m 39s | 5.63s |
| $\text{optimizer = false}$ | 30.47s | 1.85s |

### 2. Building both `src` and `test`

| Build time | $\text{ir = true}$ | $\text{ir = false}$ |
| - | - | - |
| $\text{optimizer = true}$ | 1h 10m 5s | 2m 47s |
| $\text{optimizer = false}$ | 23m 6s | 80.54s |


With unoptimised IR, the repo build time is ~23 mins. Considering the lite profile is only used during development and we need to build both `src` and `test`, I think `via_ir = false` is fine even if the generated bytecodes are different. Wdyt?

**Reply by PaulRBerg** [reply]
Exactly what I wanted to see. Thanks for running these checks.

I find it obvious to stick with our existing set-up and NOT use via IR locally. I will let Cameel know.
